tests/rptest/services/http_server.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import json
+import os
+import signal
+
+from ducktape.cluster.remoteaccount import RemoteCommandError
+from ducktape.services.background_thread import BackgroundThreadService
+
+
+class HttpServer(BackgroundThreadService):
+    """
+    Service wrapping simple HTTP server that logs requests to stdout
+    """
+
+    PERSISTENT_ROOT = "/mnt/http_server"
+    SCRIPT = os.path.join(PERSISTENT_ROOT, "server.py")
+    STDOUT_CAPTURE = os.path.join(PERSISTENT_ROOT, "http_server.stdout")
+
+    logs = {
+        "verifiable_consumer_stdout": {
+            "path": STDOUT_CAPTURE,
+            "collect_default": True
+        },
+    }
+
+    def __init__(self, context, port=8080, stop_timeout_sec=2):
+        super(HttpServer, self).__init__(context, 1)
+        self.port = port
+        self.stop_timeout_sec = stop_timeout_sec
+        self.requests = []
+        # server is running on single node
+        self.url = f"http://{self.nodes[0].account.hostname}:{self.port}"
+
+    def _worker(self, idx, node):
+        node.account.ssh("mkdir -p %s" % HttpServer.PERSISTENT_ROOT,
+                         allow_fail=False)
+
+        server_script = self.render('http_server.py')
+        node.account.create_file(HttpServer.SCRIPT, server_script)
+
+        cmd = f"python3 {HttpServer.SCRIPT} --port {self.port}"
+        cmd += f" | tee {HttpServer.STDOUT_CAPTURE} &"
+
+        self.logger.debug("Starting HTTP server %s", self.url)
+        for line in node.account.ssh_capture(cmd):
+            request = self.try_parse_json(node, line.strip())
+            if request is not None:
+                self.logger.debug("received request: %s", request)
+                self.requests.append(request)
+
+    def pids(self, node):
+        try:
+            cmd = "ps ax | grep server.py | grep -v grep | awk '{print $1}'"
```

Comment:
maybe select something more unique than `server.py` -- in principle more than one background server could run at a time (IIRC)